src/DssDirectDepositAaveDai.sol

```diff
@@ -216,25 +225,53 @@ contract DssDirectDepositAaveDai {
 
         emit File(what, data);
     }
+
     function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositAaveDai/no-file-during-shutdown");
+
         if (what == "king") king = data;
         else revert("DssDirectDepositAaveDai/file-unrecognized-param");
         emit File(what, data);
     }
 
-    // --- Deposit controls ---
-    function wind(uint256 amount) external auth {
-        require(live == 1, "DssDirectDepositAaveDai/not-live");
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) public view returns (uint256) {
+        require(targetInterestRate > 0, "DssDirectDepositAaveDai/target-interest-zero");
+        require(targetInterestRate <= interestStrategy.getMaxVariableBorrowRate(), "DssDirectDepositAaveDai/above-max-interest");
 
-        _wind(amount);
+        // Do inverse calculation of interestStrategy
+        uint256 targetUtil;
+        if (targetInterestRate > _add(interestStrategy.baseVariableBorrowRate(), interestStrategy.variableRateSlope1())) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - interestStrategy.baseVariableBorrowRate() - interestStrategy.variableRateSlope1();
+            targetUtil = _add(_rdiv(_rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r), interestStrategy.variableRateSlope2()), interestStrategy.OPTIMAL_UTILIZATION_RATE());
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, interestStrategy.baseVariableBorrowRate()), interestStrategy.OPTIMAL_UTILIZATION_RATE()), interestStrategy.variableRateSlope1());
+        }
+        return _rdiv(_add(stableDebt.totalSupply(), variableDebt.totalSupply()), targetUtil);
     }
+
+    // --- Deposit controls ---
     function _wind(uint256 amount) internal {
+        bytes32 ilk_ = ilk;
+
+        // Wind amount is limited by the debt ceiling
+        (uint256 Art,,, uint256 line,) = vat.ilks(ilk_);
+        uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+        uint256 newDebt = _add(Art, amount);
```

Comment:
In order to help future maintainers avoid hard-to-see bugs, it is important to make our assumptions explicit.

Here, we are assuming `rate = 1`, therefore normalized debt equals debt (ignoring units), therefore this addition makes sense.

Note also that computing a WAD line is uncommon in our codebase. Why is that ? Because usually we compare it to `art * rate`, which is a RAD. So, in a way, when we divide line by `RAY`, we are actually dividing it by `rate`.

My main concern is that new développers look at this and don’t see anything unusual, and then miss the distinction between normalized debt and régular debt where it matters.

I would add a comment explaining this assumption. Or, better still, I would refactor the code so that debt is measured in `RAD`, as is the practice in our core modules.

Reply:
I've added some comments, let me know what you think and if we should expand better and in which places.

Reply:
I like them, thank you !